js/src/forum/components/DiscussionListPane.js

```diff
@@ -39,20 +40,28 @@ export default class DiscussionListPane extends Component {
     // If the discussion we are viewing is listed in the discussion list, then
     // we will make sure it is visible in the viewport – if it is not we will
     // scroll the list down to it.
-    const $discussion = $list.find('.DiscussionListItem.active');
-    if ($discussion.length) {
-      const listTop = $list.offset().top;
-      const listBottom = listTop + $list.outerHeight();
-      const discussionTop = $discussion.offset().top;
-      const discussionBottom = discussionTop + $discussion.outerHeight();
+    // When coming from another discussion, scroll to the previous postition
+    // to prevent the discussion list jumping around.
+    if (!app.previous.matches(DiscussionPage)) {
```

Comment:
Can we swap this around so that the DiscussionPage special case comes first? That way we avoid negation in the conditional